urwid/main_loop.py

```diff
@@ -1324,7 +1324,10 @@ def run(self):
         self._loop.set_exception_handler(self._exception_handler)
         self._loop.run_forever()
         if self._exc_info:
-            raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
+            if PYTHON3:
+                raise self._exc_info
+            else:
+                raise self._exc_info[0](self._exc_info[1]).with_traceback(self._exc_info[2])
```

Comment:
Okay, I give up. I can't fix this without proper test cases and all tests are
broken in py3 because of issue #110, which I don't understand (what the hell
is '__super'?), so I'm stuck.